sceptre/connection_manager.py

```diff
@@ -40,13 +44,21 @@ def _retry_boto_call(func):
     def decorated(*args, **kwargs):
         max_retries = 30
         attempts = 1
+        mdelay = 1
+        delay_cap = 45
         while attempts < max_retries:
             try:
                 return func(*args, **kwargs)
             except ClientError as e:
                 if e.response["Error"]["Code"] == "Throttling":
-                    logger.error("Request limit exceeded, pausing...")
-                    time.sleep(attempts)
+                    logger.error("Request limit exceeded, pausing {}...".format(mdelay))
+                    time.sleep(mdelay)
+
+                    # Using De-correlated Jitter Algorithm
+                    # We are picking number between a ceiling (delay_cap) of 45 seconds and the last
+                    # delay multiplied by 2.5, rounding to two decimal places.
+                    mdelay = min(delay_cap, round((random.uniform(1, mdelay * 2.5)), 2))
```

Comment:
It doesn't look like `mdelay` is being updated to account for the attempt at all.